Branch-SDK/Branch-SDK/BNCSystemObserver.m

```diff
@@ -42,6 +42,18 @@ + (NSString *)getUniqueHardwareId:(BOOL *)isReal {
     return uid;
 }
 
++ (BOOL)adTrackingSafe {
+    Class ASIdentifierManagerClass = NSClassFromString(@"ASIdentifierManager");
+    if (ASIdentifierManagerClass) {
+        SEL sharedManagerSelector = NSSelectorFromString(@"sharedManager");
+        id sharedManager = ((id (*)(id, SEL))[ASIdentifierManagerClass methodForSelector:sharedManagerSelector])(ASIdentifierManagerClass, sharedManagerSelector);
+        SEL advertisingEnabledSelector = NSSelectorFromString(@"isAdvertisingTrackingEnabled");
+        BOOL enabled = ((BOOL (*)(id, SEL))[sharedManager methodForSelector:advertisingEnabledSelector])(sharedManager, advertisingEnabledSelector);
+        return enabled;
+    }
+    return YES;
```

Comment:
Right now, the default is YES. We only set NO if we can explicitly know that the user has opted out of ads